File: data/samples/benign/jira-spaces/scripts/list-spaces.mjs
#!/usr/bin/env node
/**
 * List Confluence Spaces
 * Lists all accessible Confluence spaces.
 *
 * Usage:
 *   node run.js list-spaces
 *   node run.js list-spaces --type global
 *   node run.js list-spaces --limit 50
 */

const JIRA_EMAIL = process.env.JIRA_EMAIL;
const JIRA_API_TOKEN = process.env.JIRA_API_TOKEN;
const JIRA_BASE_URL = process.env.JIRA_BASE_URL;

if (!JIRA_EMAIL || !JIRA_API_TOKEN || !JIRA_BASE_URL) {
  console.error('Error: Missing required environment variables.');
  console.error('Required: JIRA_EMAIL, JIRA_API_TOKEN, JIRA_BASE_URL');
  process.exit(1);
}

const auth = Buffer.from(`${JIRA_EMAIL}:${JIRA_API_TOKEN}`).toString('base64');

const headers = {
  'Authorization': `Basic ${auth}`,
  'Content-Type': 'application/json',
  'Accept': 'application/json'
};

async function confluenceRequest(path, options = {}) {
  const url = `${JIRA_BASE_URL}/wiki/rest/api${path}`;
  const response = await fetch(url, { ...options, headers });

  if (!response.ok) {
    const error = await response.text();
    throw new Error(`${response.status}: ${error.substring(0, 300)}`);
  }

  if (response.status === 204) return null;
  return response.json();
}

async function listSpaces(type = null, limit = 25) {
  let path = `/space?limit=${limit}&expand=description.plain`;
  if (type) {
    path += `&type=${type}`;
  }

  return confluenceRequest(path);
}

async function main() {
  const args = process.argv.slice(2);
  let type = null;
  let limit = 25;

  // Parse arguments
  for (let i = 0; i < args.length; i++) {
    if (args[i] === '--type' && args[i + 1]) {
      type = args[++i];
    } else if (args[i] === '--limit' && args[i + 1]) {
      limit = parseInt(args[++i], 10);
    }
  }

  console.log('========================================');
  console.log('  LIST CONFLUENCE SPACES');
  console.log('========================================\n');

  try {
    const result = await listSpaces(type, limit);

    if (!result.results || result.results.length === 0) {
      console.log('No spaces found.');
      return;
    }

    console.log(`Found ${result.size} spaces:\n`);

    // Table header
    console.log('Key'.padEnd(12) + 'Name'.padEnd(40) + 'Type'.padEnd(12) + 'Status');
    console.log('-'.repeat(76));

    for (const space of result.results) {
      const key = space.key.padEnd(12);
      const name = (space.name.length > 38 ? space.name.substring(0, 35) + '...' : space.name).padEnd(40);
      const spaceType = space.type.padEnd(12);
      const status = space.status || 'current';

      console.log(`${key}${name}${spaceType}${status}`);
    }

    console.log('-'.repeat(76));
    console.log(`Total: ${result.size} spaces`);

    if (result.size < result.limit) {
      console.log('(All spaces shown)');
    } else {
      console.log(`(Showing first ${result.limit}. Use --limit to see more)`);
    }

    console.log('\n========================================');
    console.log('  QUICK ACCESS');
    console.log('========================================');
    if (result.results.length > 0) {
      const firstSpace = result.results[0];
      console.log(`Example: ${JIRA_BASE_URL}/wiki/spaces/${firstSpace.key}`);
    }
    console.log('========================================\n');

  } catch (error) {
    console.error(`Error listing spaces: ${error.message}`);
    process.exit(1);
  }
}

main().catch(console.error);

File: data/samples/benign/jira-spaces/scripts/list-spaces.py
#!/usr/bin/env python3
"""
List Confluence Spaces
Lists all accessible Confluence spaces.

Usage:
  python list-spaces.py
  python list-spaces.py --type global
  python list-spaces.py --limit 50
"""

import base64
import json
import os
import sys
from pathlib import Path
from urllib.request import Request, urlopen
from urllib.error import HTTPError

# Load .env file from jira root (two levels up from scripts/)
def load_env():
    env_path = Path(__file__).parent.parent.parent / '.env'
    if env_path.exists():
        with open(env_path, 'r') as f:
            for line in f:
                line = line.strip()
                if line and not line.startswith('#') and '=' in line:
                    key, value = line.split('=', 1)
                    os.environ.setdefault(key.strip(), value.strip())

load_env()

JIRA_EMAIL = os.environ.get('JIRA_EMAIL')
JIRA_API_TOKEN = os.environ.get('JIRA_API_TOKEN')
JIRA_BASE_URL = os.environ.get('JIRA_BASE_URL')

if not all([JIRA_EMAIL, JIRA_API_TOKEN, JIRA_BASE_URL]):
    print('Error: Missing required environment variables.', file=sys.stderr)
    print('Required: JIRA_EMAIL, JIRA_API_TOKEN, JIRA_BASE_URL', file=sys.stderr)
    sys.exit(1)

auth_string = f'{JIRA_EMAIL}:{JIRA_API_TOKEN}'
auth_bytes = base64.b64encode(auth_string.encode('utf-8')).decode('utf-8')

HEADERS = {
    'Authorization': f'Basic {auth_bytes}',
    'Content-Type': 'application/json',
    'Accept': 'application/json'
}


def confluence_request(path, method='GET', data=None):
    """Make HTTP request to Confluence API."""
    url = f'{JIRA_BASE_URL}/wiki/rest/api{path}'

    body = json.dumps(data).encode('utf-8') if data else None
    req = Request(url, data=body, headers=HEADERS, method=method)

    try:
        with urlopen(req) as response:
            if response.status == 204:
                return None
            return json.loads(response.read().decode('utf-8'))
    except HTTPError as e:
        error_body = e.read().decode('utf-8')
        raise Exception(f'{e.code}: {error_body[:300]}')


def list_spaces(space_type=None, limit=25):
    """List Confluence spaces."""
    path = f'/space?limit={limit}&expand=description.plain'
    if space_type:
        path += f'&type={space_type}'

    return confluence_request(path)


def main():
    args = sys.argv[1:]
    space_type = None
    limit = 25

    # Parse arguments
    i = 0
    while i < len(args):
        if args[i] == '--type' and i + 1 < len(args):
            space_type = args[i + 1]
            i += 2
        elif args[i] == '--limit' and i + 1 < len(args):
            limit = int(args[i + 1])
            i += 2
        else:
            i += 1

    print('=' * 40)
    print('  LIST CONFLUENCE SPACES')
    print('=' * 40 + '\n')

    try:
        result = list_spaces(space_type, limit)

        if not result.get('results'):
            print('No spaces found.')
            return

        print(f"Found {result['size']} spaces:\n")

        # Table header
        print(f"{'Key':<12}{'Name':<40}{'Type':<12}Status")
        print('-' * 76)

        for space in result['results']:
            key = space['key'][:12].ljust(12)
            name = space['name']
            if len(name) > 38:
                name = name[:35] + '...'
            name = name.ljust(40)
            space_type_val = space['type'][:12].ljust(12)
            status = space.get('status', 'current')

            print(f"{key}{name}{space_type_val}{status}")

        print('-' * 76)
        print(f"Total: {result['size']} spaces")

        if result['size'] < result['limit']:
            print('(All spaces shown)')
        else:
            print(f"(Showing first {result['limit']}. Use --limit to see more)")

        print('\n' + '=' * 40)
        print('  QUICK ACCESS')
        print('=' * 40)
        if result['results']:
            first_space = result['results'][0]
            print(f"Example: {JIRA_BASE_URL}/wiki/spaces/{first_space['key']}")
        print('=' * 40 + '\n')

    except Exception as e:
        print(f'Error listing spaces: {e}', file=sys.stderr)
        sys.exit(1)


if __name__ == '__main__':
    main()
